SteamKit2/SteamKit2/Steam/Handlers/SteamUnifiedMessages/SteamUnifiedMessages.cs

```diff
@@ -64,14 +94,24 @@ public AsyncJob<ServiceMethodResponse> SendMessage<TResponse>( Expression<Func<T
                     throw new NotSupportedException( "Unknown Expression type" );
                 }
 
-                var serviceName = typeof(TService).Name.Substring( 1 ); // IServiceName - remove 'I'
+                var serviceName = typeof( TService ).Name.Substring( 1 ); // IServiceName - remove 'I'
                 var methodName = methodInfo.Name;
                 var version = 1;
 
                 var rpcName = string.Format( "{0}.{1}#{2}", serviceName, methodName, version );
 
-                var method = typeof(SteamUnifiedMessages).GetMethod( nameof(SteamUnifiedMessages.SendMessage) )!.MakeGenericMethod( message.GetType() );
-                var result = method.Invoke( this.steamUnifiedMessages, new[] { rpcName, message, isNotification } )!;
+                if( isNotification )
+                {
+                    var notification = typeof( SteamUnifiedMessages ).GetMethod( nameof( SteamUnifiedMessages.SendNotification ) )!.MakeGenericMethod( message.GetType() );
+                    notification.Invoke( this.steamUnifiedMessages, new[] { rpcName, message } );
+                    return null;
+                }
+
+                // When the obsolete overload is removed, then this code be used, instead of GetMethods().
+                // var method = typeof( SteamUnifiedMessages ).GetMethod( nameof( SteamUnifiedMessages.SendMessage ) )!.MakeGenericMethod( message.GetType() );
+
+                var method = typeof( SteamUnifiedMessages ).GetMethods().First( x => x.Name == nameof( SteamUnifiedMessages.SendMessage ) ).MakeGenericMethod( message.GetType() );
```

Comment:
`First` isn't necessarily safe here.

@JustArchi can you please retest with this?:

```csharp
var method = sendMessageMethod.MakeGenericMethod( message.GetType() );
```

and then outside the method definition:

```csharp
static readonly MethodInfo sendMessageMethod = typeof( SteamUnifiedMessages )
    .GetMethods( BindingFlags.Public | BindingFlags.Instance )
    .Single( m => m is { Name: nameof( SteamUnifiedMessages.SendMessage ) } && !Attribute.IsDefined( m, typeof( ObsoleteAttribute ) ) );
```

Reply:
Changed it to this, it works.